### PR TITLE
fix/stormshield_sns_domain_field

### PR DESCRIPTION
**Pull request by clement-burtscher-sekoia**
## Description

Relates to https://github.com/SekoiaLab/integration/issues/1935

### Added

- Parse new ECS fields:
  - `destination.geo.country_iso_code` from `dstcountry` raw field
  - `dns.question.name` from `dstname` raw field when value is domain-like and not an IP address
  - `dns.question.registered_domain` from `domain` raw field when value is non-empty
  - `source.geo.country_iso_code` from `srccountry` raw field
  - `source.port` from `port` raw field for `xvpn` logs when `srcport` is absent
- Add SSL test coverage to validate IoC-oriented domain normalization and prevent IP values from populating `dns.question.name`

### Changed

- Harmonize test fixture file naming for consistency and clearer scenario intent across the format

### Fixed

- Anonymize all test fixtures with RFC 5737 TEST-NET IP ranges and example domains

## Type of Change

- [ ] New intake format
- [ ] Update to existing intake format
- [x] Bug fix
- [ ] Documentation update
- [ ] Other (please describe):

## Affected Intake Formats

- Vendor/product: Stormshield/stormshield_network_security

### 🔒 CRITICAL - Security Requirements (MUST BE VERIFIED FIRST)

- [ ] **No real email addresses** in test files (use: user@example.com, test@example.org)
- [ ] **No real passwords or API keys** (use: "P@ssw0rd!", "xxxxxxxxxxxx", "FAKE_TOKEN_123")
- [ ] **No real IP addresses** (use TEST-NET ranges: 192.0.2.x, 198.51.100.x, 203.0.113.x)
- [ ] **No real phone numbers** (use: +1-555-0100 or similar test numbers)
- [ ] **No real names or PII** (use: "John Doe", "Jane Smith", "User123")
- [ ] **No real company/organization names** in test data (unless public vendors)
- [ ] **No real usernames, hostnames, or domain names** (use: user1, host.example.com, test.corp)
- [ ] All test data has been anonymized and verified

### Required for All PRs

- [ ] Code has been linted
    - [ ] with the linter for manifest, smart-descriptions, and test files (`uv run python linter.py check --changes` in utils/)
    - [ ] with Prettier for the parser code files (`npx prettier --write <module>/<format>/ingest/parser.yml`)
- [ ] All CI/CD checks pass
- [ ] Changes follow the contribution guidelines in `CONTRIBUTING.md`

### Required for New/Updated Intake Formats

- [ ] Clear descriptions provided for modules and formats
- [ ] Logo files included for new modules/formats
- [ ] Parser test coverage is at least 75%
- [ ] At least one smart-description is provided
- [ ] README.md updated (if applicable)
- [ ] Taxonomy mappings are correct and documented
- [ ] Test cases cover edge cases and error handling

### Testing

- [ ] Local tests pass (`uv run pytest` in utils/)
- [ ] Sample data is representative and **completely anonymized**
- [ ] **No sensitive information** (PII, credentials, real IPs, real emails) in test files
- [ ] Test data uses example.com/example.org domains and TEST-NET IP ranges
- [ ] Parser handles malformed data gracefully

### Documentation

- [ ] Code changes are documented
- [ ] Smart-descriptions are clear and informative
- [ ] Field mappings are explained

## Related Issues

Closes #

## Summary by Sourcery

Improve Stormshield SNS ECS mappings and SSL domain handling while expanding and standardizing parser test coverage.

New Features:
- Map Stormshield SNS country, DNS, and VPN source-port data to the corresponding ECS fields.

Bug Fixes:
- Prevent IP addresses and unsuitable values from being populated as DNS question names while normalizing SSL event domains.
- Anonymize test fixtures with safe example domains and TEST-NET addresses.

Enhancements:
- Standardize test fixture naming to make scenarios clearer and more consistent.

Tests:
- Add SSL parser coverage for domain normalization and DNS field extraction.

### CODE REVIEW

File: Stormshield/stormshield_network_security/tests/test_xvpn_03.json [hunk 1]
```diff
@@ -48,7 +48,8 @@
     },
     "source": {
       "address": "1.2.3.4",
-      "ip": "1.2.3.4"
+      "ip": "1.2.3.4",
+      "port": 59505
     },
```
[the remark this answers is not in the document]
Comment (clement-burtscher-sekoia) [reply Stormshield/stormshield_network_security/tests/test_xvpn_03.json:53]: Fixed with commit [Fix tests anonymization](https://github.com/SEKOIA-IO/intake-formats/pull/2888/changes/ca59e1b6aa4607081f5fc2c18751931c23bb0b5e)

File: Stormshield/stormshield_network_security/tests/test_filter_02.json [hunk 2]
```diff
@@ -24,7 +24,8 @@
     "destination": {
       "address": "22.22.22.22",
       "geo": {
-        "continent_name": "eu"
+        "continent_name": "eu",
+        "country_iso_code": "be"
```
[the remark this answers is not in the document]
Comment (clement-burtscher-sekoia) [reply Stormshield/stormshield_network_security/tests/test_filter_02.json:28]: Fixed with commit [Fix tests anonymization](https://github.com/SEKOIA-IO/intake-formats/pull/2888/changes/ca59e1b6aa4607081f5fc2c18751931c23bb0b5e)

File: Stormshield/stormshield_network_security/tests/filter.json [hunk 3]
```diff
@@ -25,7 +25,8 @@
     "destination": {
       "address": "11.11.11.11",
       "geo": {
-        "continent_name": "na"
+        "continent_name": "na",
+        "country_iso_code": "us"
       },
       "ip": "11.11.11.11",
```
[the remark this answers is not in the document]
Comment (clement-burtscher-sekoia) [reply Stormshield/stormshield_network_security/tests/filter.json]: Fixed with commit [Fix tests anonymization](https://github.com/SEKOIA-IO/intake-formats/pull/2888/changes/ca59e1b6aa4607081f5fc2c18751931c23bb0b5e)

File: Stormshield/stormshield_network_security/tests/test_empty_action_01.json [hunk 4]
```diff
@@ -25,7 +25,8 @@
     "destination": {
       "address": "11.22.33.44",
       "geo": {
-        "continent_name": "na"
+        "continent_name": "na",
+        "country_iso_code": "us"
```
[the remark this answers is not in the document]
Comment (clement-burtscher-sekoia) [reply Stormshield/stormshield_network_security/tests/test_empty_action_01.json:29]: Fixed with commit [Fix tests anonymization](https://github.com/SEKOIA-IO/intake-formats/pull/2888/changes/ca59e1b6aa4607081f5fc2c18751931c23bb0b5e)

File: Stormshield/stormshield_network_security/tests/test_plugin_01.json [hunk 5]
```diff
@@ -25,7 +25,8 @@
     "destination": {
       "address": "55.66.77.88",
       "geo": {
-        "continent_name": "eu"
+        "continent_name": "eu",
+        "country_iso_code": "fr"
```
[the remark this answers is not in the document]
Comment (clement-burtscher-sekoia) [reply Stormshield/stormshield_network_security/tests/test_plugin_01.json:29]: Fixed with commit [Fix tests anonymization](https://github.com/SEKOIA-IO/intake-formats/pull/2888/changes/ca59e1b6aa4607081f5fc2c18751931c23bb0b5e)

File: Stormshield/stormshield_network_security/ingest/parser.yml [hunk 6]
```diff
@@ -603,6 +609,14 @@ stages:
           url.path: "{{ kv.result.arg }}"
         filter: '{{ kv.result.logtype == "plugin" and kv.result.proto == "http" }}'
 
+      - set:
+          dns.question.name: "{{ kv.result.dstname }}"
+        filter: '{{ kv.result.logtype == "ssl" and kv.result.get("dstname") and "." in kv.result.dstname and "_" not in kv.result.dstname }}'
```
[the remark this answers is not in the document]
Comment (clement-burtscher-sekoia) [reply Stormshield/stormshield_network_security/ingest/parser.yml]: Fixed with commit [Apply changes requested by copilot review](https://github.com/SEKOIA-IO/intake-formats/pull/2888/changes/e2da136db9119c2b0d610e95569b4813cab7d2da)

File: Stormshield/stormshield_network_security/_meta/smart-descriptions.json [hunk 7]
```diff
@@ -311,6 +311,9 @@
       {
         "field": "source.ip"
       },
+      {
+        "field": "source.port"
+      },
```
[the remark this answers is not in the document]
Comment (clement-burtscher-sekoia) [reply Stormshield/stormshield_network_security/_meta/smart-descriptions.json]: Fixed with commit [Apply changes requested by copilot review](https://github.com/SEKOIA-IO/intake-formats/pull/2888/changes/e2da136db9119c2b0d610e95569b4813cab7d2da)

File: Stormshield/stormshield_network_security/tests/ssl_domain_ioc_dstname.json [hunk 8]
```diff
@@ -0,0 +1,100 @@
+{
+  "input": {
+    "message": "id=firewall time=\"2026-09-04 12:12:56\" fw=\"SN12345678912345\" tz=+0200 startime=\"2026-09-04 12:12:55\" pri=5 proto=ssl confid=1 slotlevel=2 ruleid=124 rulename=\"decrypt\" user=\"\" domain=\"\" contentpolicy=1 cnruleid=13 src=192.0.2.179 srcport=55127 srcportname=ephemeral_fw_tcp dst=198.51.100.30 dstport=443 dstportname=https srcmac=aa:bb:cc:dd:ee:ff modsrc=203.0.113.9 modsrcport=13836 origdst=198.51.100.30 origdstport=443 ipv=4 sent=2102 rcvd=5285 duration=0.24 action=pass dstname=\"portal.example.com\" msg=\"Connection not deciphered (direct rule matches, no decipher)\" cat_site=\"phishing\" srchostrep=340 dstcontinent=\"na\" dstcountry=\"ca\" dstiprep=\"sekoiafqdn\" logtype=\"ssl\""
```
[the remark this answers is not in the document]
Comment (clement-burtscher-sekoia) [reply Stormshield/stormshield_network_security/tests/ssl_domain_ioc_dstname.json]: Fixed with commit [Apply changes requested by copilot review](https://github.com/SEKOIA-IO/intake-formats/pull/2888/changes/72c8bdbcf951d3403ba61af7b1b59221fe90a822)

File: Stormshield/stormshield_network_security/tests/ssl_domain_ioc_registered_domain.json [hunk 9]
```diff
@@ -0,0 +1,103 @@
+{
+  "input": {
+    "message": "id=firewall time=\"2026-09-04 12:12:56\" fw=\"SN12345678912345\" tz=+0200 startime=\"2026-09-04 12:12:55\" pri=5 proto=ssl confid=1 slotlevel=2 ruleid=124 rulename=\"decrypt\" user=\"\" domain=\"example.com\" contentpolicy=1 cnruleid=13 src=192.0.2.179 srcport=55127 srcportname=ephemeral_fw_tcp dst=198.51.100.30 dstport=443 dstportname=https srcmac=aa:bb:cc:dd:ee:ff modsrc=203.0.113.9 modsrcport=13836 origdst=198.51.100.30 origdstport=443 ipv=4 sent=2102 rcvd=5285 duration=0.24 action=pass dstname=\"portal.example.com\" msg=\"Connection not deciphered (direct rule matches, no decipher)\" cat_site=\"phishing\" srchostrep=340 dstcontinent=\"na\" dstcountry=\"ca\" dstiprep=\"sekoiafqdn\" logtype=\"ssl\""
```
[the remark this answers is not in the document]
Comment (clement-burtscher-sekoia) [reply Stormshield/stormshield_network_security/tests/ssl_domain_ioc_registered_domain.json]: Fixed with commit [Apply changes requested by copilot review](https://github.com/SEKOIA-IO/intake-formats/pull/2888/changes/72c8bdbcf951d3403ba61af7b1b59221fe90a822)

File: Stormshield/stormshield_network_security/tests/ssl_domain_ioc_dstname_ip.json [hunk 10]
```diff
@@ -0,0 +1,91 @@
+{
+  "input": {
+    "message": "id=firewall time=\"2026-09-04 12:12:56\" fw=\"SN12345678912345\" tz=+0200 startime=\"2026-09-04 12:12:55\" pri=5 proto=ssl confid=1 slotlevel=2 ruleid=124 rulename=\"decrypt\" user=\"\" domain=\"\" contentpolicy=1 cnruleid=13 src=192.0.2.179 srcport=55127 srcportname=ephemeral_fw_tcp dst=198.51.100.30 dstport=443 dstportname=https srcmac=aa:bb:cc:dd:ee:ff modsrc=203.0.113.9 modsrcport=13836 origdst=198.51.100.30 origdstport=443 ipv=4 sent=2102 rcvd=5285 duration=0.24 action=pass dstname=\"198.51.100.130\" msg=\"Connection not deciphered (direct rule matches, no decipher)\" cat_site=\"phishing\" srchostrep=340 dstcontinent=\"na\" dstcountry=\"ca\" dstiprep=\"sekoiafqdn\" logtype=\"ssl\""
```
[the remark this answers is not in the document]
Comment (clement-burtscher-sekoia) [reply Stormshield/stormshield_network_security/tests/ssl_domain_ioc_dstname_ip.json]: Fixed with commit [Apply changes requested by copilot review](https://github.com/SEKOIA-IO/intake-formats/pull/2888/changes/72c8bdbcf951d3403ba61af7b1b59221fe90a822)

**Review by mchupeau-sk — APPROVED** [review]
LGFM

**Comment by clement-burtscher-sekoia** [comment]
image
The reviewer of this PR
